include/uio.h

```diff
@@ -21,6 +21,19 @@ typedef struct uio {
   vm_map_t *uio_vmspace; /* destination address space */
 } uio_t;
 
+#define UIO_SINGLE(op, vm_map, offset, buf, buflen)                            \
+  (uio_t) {                                                                    \
+    .uio_iov = (iovec_t[1]){(iovec_t){(buf), (buflen)}}, .uio_iovcnt = 1,      \
```

Comment:
The space for `iov` gets allocated on stack. With C99 (checked with `-std=c99 -Wpedantic`) you can do following:
```c
void foobar() {
  char *str;
  str = "foobar";
  str = (char[6]){'h', 'e', 'l', 'l', 'o', 0};
  int *tab;
  tab = (int[4]){1,2,3,4};
}
```
In both cases arrays (and string as well) will be allocated on stack (checked with optimizations turned off and `objdump -d`), unless they are constant (if so they're put into `.rodata` section). Does this explanation help?